api/v1alpha1/hostedclusterinfrastructure_types.go

```diff
@@ -0,0 +1,341 @@
+	CloudProvider corev1.LocalObjectReference `json:"cloudProvider,omitempty"`
+}
+
+// PlatformSpec specifies the underlying infrastructure provider for the cluster
+// and is used to configure platform specific behavior.
+type PlatformInfraSpec struct {
+	// Type is the type of infrastructure provider for the cluster.
+	//
+	// +kubebuilder:validation:Enum=AWS;Azure;PowerVS
+	// +immutable
+	Type PlatformType `json:"type"`
+
+	// AWS specifies configuration for clusters running on Amazon Web Services.
+	//
+	// +optional
+	// +immutable
+	AWS *AWSPlatformInfraSpec `json:"aws,omitempty"`
+
+	// Azure defines azure specific settings
+	Azure *AzurePlatformSpec `json:"azure,omitempty"`
+
+	// PowerVS specifies configuration for clusters running on IBMCloud Power VS Service.
+	// This field is immutable. Once set, It can't be changed.
+	//
+	// +optional
+	// +immutable
+	PowerVS *PowerVSPlatformSpec `json:"powervs,omitempty"`
+}
+
+// AWSPlatformSpec specifies configuration for clusters running on Amazon Web Services.
+type AWSPlatformInfraSpec struct {
+	// Region is the AWS region in which the cluster resides. This configures the
+	// OCP control plane cloud integrations, and is used by NodePool to resolve
+	// the correct boot AMI foHostedClusterInfrastructureConfigStatus a given release.
+	// HostedCluster.spec.platform.aws.region
+	//
+	// +immutable
+	Region string `json:"region"`
+}
+
+// HostedClusterInfrastructureConfigStatus reports the IDs of the platform resources via a status field
+type HostedClusterInfrastructureConfigStatus struct {
+
+	// Platform specifies the underlying infrastructure provider for the cluster
+	// and is used to configure platform specific behavior.
+	//
+	// +immutable
+	Platform PlatformInfraConfigStatus `json:"platform"`
+
+	// DNS specifies DNS configuration for the cluster.
+	//
+	// +immutable
+	DNS DNSSpec `json:"dns"`
+
+	// Networking specifies network configuration for the cluster.
+	//
+	// +optional
+	// +immutable
+	Networking ClusterNetworkingInfraConfigStatus `json:"networking,omitempty"`
+
+	// IssuerURL is an OIDC issuer URL which is used as the issuer in all
+	// ServiceAccount tokens generated by the control plane API server. The
+	// default value is kubernetes.default.svc, which only works for in-cluster
+	// validation.
+	//
+	// +kubebuilder:default:="https://kubernetes.default.svc"
+	// +immutable
+	// +optional
+	// +kubebuilder:validation:Format=uri
+	IssuerURL string `json:"issuerURL,omitempty"`
+}
+
+type ClusterNetworkingInfraConfigStatus struct {
+	// MachineNetwork is the list of IP address pools for machines.
+	// TODO: make this required in the next version of the API
+	//
+	// +immutable
+	// +optional
+	MachineNetwork []MachineNetworkEntry `json:"machineNetwork,omitempty"`
+}
+
+// PlatformInfraConfigStatus specifies the underlying infrastructure provider for the cluster
+// and is used to configure platform specific behavior.
+type PlatformInfraConfigStatus struct {
```

Comment:
PowerVS?

Reply:
I'm not aware of what infrastructure steps we do in PowerVS, so it would be added later like Azure which is present but isn't actionable.

Reply:
OK